explainaboard/metric.py

```diff
@@ -1,172 +1,373 @@
-from random import choices
+from __future__ import annotations
+
+import abc
+from collections import Counter
+import itertools
+import re
+import string
+import sys
+from typing import Any, Optional, Union
 
 import numpy as np
-import scipy
-import sklearn.metrics
-from sklearn.metrics import accuracy_score
+
+from explainaboard.utils.async_eaas import AsyncEaaSResult
+
+
+class MetricResult:
+    def __init__(
+        self,
+        name: str,
+        value: float,
+        conf_interval: Optional[tuple[float, float]] = None,
+        conf_value: Optional[float] = None,
+    ):
+        """Initialize a result of an evaluation metric
+
+        :param name: name of the metric
+        :param value: value of the metric
+        :param conf_interval: the confidence interval of the metric
+        :param conf_value: the p-value of the confidence interval
+        """
+        self.name = name
+        self.value = value
+        self.conf_interval = conf_interval
+        self.conf_value = conf_value
```

Comment:
If these values are unchanged once they are initialized, would it be reasonable to use frozen dataclasses or tag them with the `Final` type hint? it is generally better to avoid exposing mutable members from usual classes.
If exposing the members is intended for some reason, could you provide details of the usage and appropriate type hints (since these members are typeless, users may insert any kinds of data after initializing the struct)?

ditto for other classes.